ci/prow/e2e-crd-variable

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+echo "Deploy KMMO..."
+make deploy
+
+echo "Wait until the KMMO Deployment is Available and "
+echo "Describe the Deployment and get its YAML if that failed..."
+timeout 1m kubectl wait --for condition=Available deployments.apps -n kmm-operator-system kmm-operator-controller-manager || kubectl describe deployments.apps -n kmm-operator-system kmm-operator-controller-manager && kubectl get -o yaml deployments.apps -n kmm-operator-system kmm-operator-controller-manager
+
+echo "Check that the kmm_ci_a module is not loaded on the node..."
+if minikube ssh -- lsmod | grep kmm_ci_a; then
+   echo "Unexpected lsmod output - the module should not be loaded"
+   exit 1
+fi
+
+echo " Add an kmm-ci Module that contains a valid mapping..."
+kubectl apply -f ci/module-kmm-ci-variable.yaml
+
+echo "Check that the module gets loaded on the node..."
+timeout 1m bash -c 'until minikube ssh -- lsmod | grep kmm_ci_a; do sleep 3; done'
+
+echo "Remove the Module..."
+kubectl delete -f ci/module-kmm-ci-variable.yaml
+
+echo "Check that the module gets unloaded from the node..."
+timeout 1m bash -c 'until ! minikube ssh -- lsmod | grep kmm_ci_a; do sleep 3; done'
+
+echo "Collect troubleshooting data..."
+./ci/github/collect_troubleshooting
```

Comment:
Won't it collect only on success?

Reply:
**Also
(can't update the original comment for some reason)

Reply:
We collect on success today.

Reply:
Any reason to not do it only on failure if we are changing the structure of the tests?

Reply:
Now in actions it is configured as [always()](https://github.com/kubernetes-sigs/kernel-module-management/blob/main/.github/workflows/e2e.yaml#L530), so I just tried to keep it similar.

Reply:
I think we should update it. No one will look at it in case of success, this is just more code that can fail, more CPU cycles, slower CI etc.

This is why I have suggested https://github.com/kubernetes-sigs/kernel-module-management/pull/166#discussion_r1033541627 so we don't have to have add the following to each command:
```
<some command> || <handle error>
```

Reply:
This makes sense to me. WDYT @qbarrand ?

Reply:
This currently takes one second: https://github.com/kubernetes-sigs/kernel-module-management/actions/runs/3573079866/jobs/6006759230
I would keep collecting it because it has virtually no cost and can be useful to understand issues where the script exits normally but still something isn't working as expected.
I would move it back to `e2e.yaml` though, because CI jobs calling that script might use an alternative collection method (e.g. must-gather on OCP).

Reply:
> can be useful to understand issues where the script exits normally but still something isn't working as expected

Do you have any example for that? We should not encounter such when using `set -e`.

Reply:
Yup. It should be part of the infra anyway. 

I think we still need to print only on failures. I think there is an easy option to set it in the actions if I am not mistaken.

Reply:
How about this?

> I would keep collecting it because it has virtually no cost and can be useful to understand issues where the script exits normally but still something isn't working as expected.

Let us not change anything from the current logic unless it actually causes issues.

Reply:
How will it work m/s? will you collect it twice? once in the job and another time in the must-gather artifact?

Reply:
No, because @enriquebelarte has moved the collection to an Action step. We can reuse the script m/s because it does not do any collection whatsoever.